Fix static deadlocks and fabric callback load order

Rearth opened the pull request:
This PR includes a few different chanes/updates, please let me know if you'd like me to split them into different PRs:
- Remove Energizer Power as mod implementation on neoforge, as it also starts up in dev environments that include the API (and the pinned version does not start up on 1.21.1)
- Move static fields from `FluidIngredientRegistry` to `FluidIngredientType` to prevent static initialization deadlocks (it seems like those 3 fields were all that's needed, couldn't reproduce the error since)
- On fabric, the initialization has been moved to the "SERVER_STARTING" event, as all content from other mods should be finished at that point. This feels a bit janky, but otherwise mods using CSL either have to call csl init again, and then get 100s of warning for duplicate API provider registrations, or they have a chance to be loaded after CSL and then won't work. If you have a better solution for this I'd be happy to remove my workaround.
- Add fallback initialization conditions for block entities. The actual check implemenation is moved to the platform implementations, as on fabric a mixin is needed, and I have no idea how to do that in the common project.

I was not able to boot up CSL directly from the IDE, and instead tested these changes by building it and loading the files in my mod Oritech, as seen here: https://github.com/Rearth/Oritech/tree/feature/architectury. The changes worked as expected on both neoforge and fabric.

Rearth committed:
Remove energized power dependency

Rearth committed:
Fix fabric entrypoint

Rearth committed:
Fix fluid Ingredient static deadlock

Rearth committed:
Bump version

Rearth committed:
Bump version to 0.0.6

Rearth committed:
Move ingredient types map to type class to avoid static deadlocks

Rearth committed:
Initialize callback on server startup on fabric

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/terrarium-earth/Common-Storage-Lib?pullRequest=38) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/terrarium-earth/Common-Storage-Lib?pullRequest=38) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/terrarium-earth/Common-Storage-Lib?pullRequest=38) it.</sub>

Rearth committed:
Only register fallbacks for blockentities implementing providers

Rearth committed:
Stupid access wideners not working

Rearth committed:
Remove interface checks again

Rearth committed:
Cleanup

Rearth commented:
Alright I removed the fallback checks for now, as I have no idea how to get the access widener to work here. I'd like to merge this change for now so we have a working version available. I still think the callback initialization could use some work, as the current solution I added to initialize late (on fabric) isn't the most elegant one.

If I understand the issue correctly then either having the mods register the lookups themselfes (for the stuff they want) or at least checking the interface implementation as I tried [here ](https://github.com/terrarium-earth/Common-Storage-Lib/pull/38/commits/a4f66e92151620764aa09eacfde81d346a6f9a16) would be a good solution.

Rearth committed:
Add implementation checks for block entity registrations

Rearth commented:
Hi @CodexAdrian just pushed another change: A validator that checks if a block entity that a fallback is being registered for actually has block types set, and also if it implements the respective API. This is needed as on fabric other mods can register block entities without having blocks set for them (no idea why they do what). When trying to register a fallback for this block entity type, fabric crashes as that's not supported there. I have multiple reports of crashes like this one: https://mclo.gs/9StzrqZ
This change also means that fallbacks are only added for block entities that actually implement the respective interfaces.

Rearth committed:
Revert "Add implementation checks for block entity registrations"

This reverts commit 3f93cbeebd91932d4d0cc80479435745a9f2ba93.

Rearth committed:
Remove fallback initialization, add self init method for block entities

Rearth commented:
Okay creating instances of the block entity types to check if they implement the right interfaces does not work with all mods, as some expect the level to be loaded in (e.g. https://mclo.gs/eEaaK55). As other mods are failing with both the interface implementation check, I don't see a way forward with the block entity fallbacks being registered for all block entity types. It'll just lead to issues as soon a mod using CSL is put into bigger modpacks.

I now went ahead and removed the fallbacks entirely, and instead also added a selfRegistration method for block entities, so mods using CSL should just register their own stuff. I know this is a big change in how CSL will be used by other mods, and I totally understand if that's not the way forward with the lib. 

Please let me know how you want to proceed here.

CodexAdrian commented on core/common/src/main/java/earth/terrarium/common_storage_lib/CommonStorageLib.java:
Please rollback this change, I've found a different solution

Rearth replied:
Great. Just updated the PR accordingly

Rearth committed:
Re-enable fallback initialization
